cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/file/FileSetDataset.java

```diff
@@ -269,17 +269,13 @@ public void onFailure() throws DataSetException {
     }
 
     try {
-      if (outputLocation.exists()) {
-        // Only delete the configured output directory, if it is empty.
-        // On Failure, org.apache.hadoop.mapreduce.lib.output.FileOutputFormat will remove files that it wrote,
-        // but it leaves around the directory that it created.
-        // We don't want to unconditionally delete the output directory on failure, because it may have files written
-        // by a different job.
-        if (outputLocation.isDirectory() && outputLocation.list().isEmpty()) {
-          if (!outputLocation.delete()) {
-            throw new DataSetException(String.format("Error deleting file(s) at path %s.", outputLocation));
-          }
-        }
+      // Only delete the configured output directory, if it is empty.
+      // On Failure, org.apache.hadoop.mapreduce.lib.output.FileOutputFormat will remove files that it wrote,
+      // but it leaves around the directory that it created.
+      // We don't want to unconditionally delete the output directory on failure, because it may have files written
+      // by a different job.
+      if (outputLocation.isDirectory() && !outputLocation.delete()) {
+        throw new DataSetException(String.format("Error deleting file(s) at path %s.", outputLocation));
```

Comment:
I disagree with this. The use case is that there is existing data and that fails the map reduce. We should not throw an exception in that case. It will only be confusing,